### PR TITLE
IPv6 remotes support

### PR DESCRIPTION
**Pull request by shawnhwei**
I don't touch Go much but it seems like this maybe all that's needed enable IPv6 remotes (#197).

### CODE REVIEW

File: share/settings/remote.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func DecodeRemote(s string) (*Remote, error) {
 			return nil, errors.New("Missing ports")
 		}
 		if !isHost(p) {
-			return nil, errors.New("Invalid host")
+			return nil, errors.New("Invalid host" + p)
```
Comment (jpillora) [inline share/settings/remote.go:88]: this could return `fmt.Errorf("Invalid host (%s)", p)`

File: share/settings/remote.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ func DecodeRemote(s string) (*Remote, error) {
 		s = strings.TrimPrefix(s, revPrefix)
 		reverse = true
 	}
-	parts := strings.Split(s, ":")
+	parts := regexp.MustCompile(`([A-Za-z0-9./]+|\[[A-Za-z0-9:/]+\]):?`).FindAllStringSubmatch(s, -1)
```
Comment (jpillora) [inline share/settings/remote.go:49]: Is `[A-Za-z0-9./]+` all valid characters for a hostname?

Maybe put the `[hex and colons]` match on the left hand side of the OR `|` and then make the hostname matcher more lax?

File: share/settings/remote_test.go [hunk 3]
```diff
@@ -81,6 +81,16 @@ func TestRemoteDecode(t *testing.T) {
 			},
 			"localhost:5353:1.1.1.1:53/udp",
 		},
+		{
+			"[::1]:8080:google.com:80",
+			Remote{
+				LocalHost:  "[::1]",
+				LocalPort:  "8080",
+				RemoteHost: "google.com",
```
Comment (jpillora) [inline share/settings/remote_test.go:89]: I'm sure it'd work, though could you please add another test with v6 in both local and remote hosts?

File: share/settings/remote.go (hunk 1 above)
Comment (shawnhwei) [reply share/settings/remote.go]: My bad, forgot to remove this after doing some testing.

File: share/settings/remote.go (hunk 2 above)
Comment (shawnhwei) [reply share/settings/remote.go]: That's a good point, this misses at least UTF8 domains and domains with hyphens.

File: share/settings/remote_test.go (hunk 3 above)
Comment (shawnhwei) [reply share/settings/remote_test.go:98]: Done.

**Comment by jpillora** [comment]
Thanks! Merged, releasing as `v1.7.1`

**Pull request merged**